connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -147,11 +156,25 @@ public void configureRoute() throws Exception {
 
         from(direct(ENGINE_TO_CONNECTOR))
             .routeId(this.connectorConfig.getConnectorName())
-            /*
-             * Dispatches the exchanges either to fetch the users directly, or, if
-             * a group ID is present, fetch the user from that group from RBAC.
-             */
-            .process(this.dispatcherProcessor);
+            // Split each recipient setting and aggregate the usernames to end
+            // up with a single exchange.
+            .split(simpleF("${exchangeProperty.%s}", ExchangeProperty.RECIPIENT_SETTINGS), this.userAggregationStrategy).stopOnException()
+                // Initialize the usernames hash set, where we will gather the
+                // fetched users from the user providers.
+                .process(exchange -> exchange.setProperty(ExchangeProperty.USERNAMES, new HashSet<String>()))
```

Comment:
Shouldn't this be done before the `split` block begins?

Reply:
I think it doesn't matter:

* If we set the `USERNAMES` property before the split, all the split exchanges will have it when the split block begins.
* The way it is right now, every exchange starts without it but we add it afterwards.

I put it there because the exchange property is only relevant to the split block, but maybe it is a good idea to reduce the operations need to be done.

Reply:
You're right, it doesn't really matter.

Reply:
I've changed it anyways, that way we reduce that operation to being executed once.